src/Papyros.ts

```diff
@@ -215,9 +177,45 @@ export class Papyros {
                 const code = getCodeForExample(papyros.codeState.programmingLanguage, name);
                 papyros.setCode(code);
             }, "input");
+            // Ensure there is no initial selection
             removeSelection(EXAMPLE_SELECT_ID);
         }
-        return papyros.launch();
+        return papyros;
+    }
+
+    async configureInput(allowReload: boolean,
+        serviceWorkerRoot?: string, serviceWorkerName?: string): Promise<boolean> {
+        const RELOAD_STORAGE_KEY = "__papyros_reloading";
+        if (allowReload && window.localStorage.getItem(RELOAD_STORAGE_KEY)) {
+            // We are the result of the page reload, so we can start
+            window.localStorage.removeItem(RELOAD_STORAGE_KEY);
+            return true;
+        } else {
+            if (typeof SharedArrayBuffer === "undefined") {
+                papyrosLog(LogType.Important, "SharedArrayBuffers are not available. ");
+                if (!serviceWorkerRoot || !serviceWorkerName || !("serviceWorker" in navigator)) {
+                    papyrosLog(LogType.Important, "Unable to register service worker. Please specify all required parameters and ensure service workers are supported.");
+                    return false;
+                }
+                if ("serviceWorker" in window.navigator) {
```

Comment:
You can remove this condition since you checked it above.

Reply:
Indeed, slight oversight in the merge.